src/lib/hash/siphash.dasl

```diff
@@ -45,23 +45,14 @@ local function finish (name, prototype, Dst)
    return ffi.cast(prototype, mcode)
 end
 
-local function random_bytes(count)
-   local bytes = ffi.new('uint8_t[?]', count)
-   local f = S.open('/dev/urandom', 'rdonly')
-   local written = 0
-   while written < count do
-      written = written + assert(f:read(bytes, count-written))
-   end
-   return bytes
-end
-
 function random_sip_hash_key()
-   return random_bytes(16)
+   return lib.random_bytes(16)
 end
 
-function reference_sip_hash_key()
-   return ffi.new('uint8_t[16]',
-                  0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15)
+function sip_hash_key_from_seed(seed)
+   local res = ffi.new('uint8_t[16]')
+   ffi.cast('uint32_t*', res)[0] = assert(tonumber(seed))
+   return res
```

Comment:
SipHash keys are 16 bytes log.  However a hash "seed" is much less -- in practice, just 4 bytes.  So we set 4 bytes of the SipHash key from the "seed", and leave the rest as zeros.